pyatlan/model/typedef.py

```diff
@@ -1,241 +1,462 @@
-        None,
-        description="Username of the user who created the object.\n",
-        example="jsmith",
+        description="Username of the user who created the object."
     )
     description: Optional[str] = Field(
-        None, description="Description of the type_ definition."
+        description="Description of the type definition."
     )
     guid: Optional[str] = Field(
-        None,
-        description="Unique identifier that represents the type_ definition.",
-        example="917ffec9-fa84-4c59-8e6c-c7b114d04be3",
-    )
-    name: str = Field(..., description="Unique name of this type_ definition.\n")
-    type_version: Optional[str] = Field(
-        None, description="Internal use only.\n", example="1.0"
+        description="Unique identifier that represents the type definition."
     )
+    name: str = Field(description="Unique name of this type definition.")
+    type_version: Optional[str] = Field(description="Internal use only.")
     update_time: Optional[int] = Field(
-        None,
-        description="Time (epoch) at which this object was last assets_updated, in milliseconds.\n",
-        example=1649172284333,
+        description="Time (epoch) at which this object was last assets_updated, in milliseconds."
     )
     updated_by: Optional[str] = Field(
-        None,
-        description="Username of the user who last assets_updated the object.\n",
-        example="jsmith",
-    )
-    version: Optional[int] = Field(
-        None, description="Version of this object.\n", example=2
+        description="Username of the user who last assets_updated the object."
     )
+    version: Optional[int] = Field(description="Version of this type definition.")
 
 
 class EnumDef(TypeDef):
     class ElementDef(AtlanObject):
-        value: str = Field(None, description="Unused.")
-        description: Optional[str] = Field(None, description="Unused.")
-        ordinal: Optional[int] = Field(None, description="Unused.")
+        value: str = Field(description="One unique value within the enumeration.")
+        description: Optional[str] = Field(description="Unused.")
+        ordinal: Optional[int] = Field(
+            description="Unique numeric identifier for the value."
+        )
+
+        @staticmethod
+        def of(ordinal: int, value: str) -> EnumDef.ElementDef:
+            from pyatlan.model.assets import validate_required_fields
+
+            validate_required_fields(
+                ["ordinal", "value"],
+                [ordinal, value],
+            )
+            return EnumDef.ElementDef(ordinal=ordinal, value=value)
+
+        @staticmethod
+        def list_from(values: List[str]) -> List[EnumDef.ElementDef]:
+            from pyatlan.model.assets import validate_required_fields
+
+            validate_required_fields(
+                ["values"],
+                [values],
+            )
+            elements = []
+            if values:
+                for i in range(0, len(values), 1):
```

Comment:
I would suggest replacing 196/197 with the following:
`elements.extend(EnumDef.ElementDef.of(ordinal=i, value=values[i]) for i in range(len(values)))`